web-ui/src/components/admin/roles/Roles.jsx

```diff
@@ -149,152 +171,183 @@ const Roles = () => {
+    setEditedRole({ ...editedRole, role: event?.target?.value });
+  }
+
+  const setRoleDescription = (event) => {
+    setEditedRole({ ...editedRole, description: event?.target?.value });
+  }
 
   return (
     <div className="roles-content">
       <div className="roles">
         <div className="roles-top">
           <div className="roles-top-left">
-            <h2>Roles</h2>
-            <TextField
-              className="role-search"
-              label="Search Roles"
-              placeholder="Role"
-              fullWidth={true}
-              value={searchText}
-              onChange={(e) => {
-                setSearchText(e.target.value);
+            <div className="roles-top-search-fields">
+              <FormControl size="small" className="role-select">
+                <InputLabel id="roles-select-label">Roles</InputLabel>
+                <Select
+                  labelId="roles-select-label"
+                  multiple
+                  value={selectedRoles}
+                  onChange={(event) => {
+                    const value = event.target.value;
+                    setSelectedRoles(typeof value === "string" ? value.split(",") : value)
+                  }}
+                  input={<OutlinedInput label="Roles"/>}
+                  renderValue={(selected) => selected.join(", ")}
+                >
+                  {roles?.map((roleObj) => (
+                    <MenuItem key={roleObj.role} value={roleObj.role}>
+                      <Checkbox checked={selectedRoles.indexOf(roleObj.role) > -1}/>
+                      <ListItemText primary={roleObj.role}/>
+                    </MenuItem>
+                  ))}
+                </Select>
+                <FormHelperText>{`Showing ${selectedRoles.length}/${roles?.length} roles`}</FormHelperText>
+              </FormControl>
+              <TextField
+                className="member-role-search"
+                label="Search members"
+                placeholder="Member Name"
+                value={searchText}
+                onChange={(e) => {
+                  setSearchText(e.target.value);
+                }}
+                InputProps={{
+                  endAdornment: <InputAdornment color="gray" position="end"><SearchIcon/></InputAdornment>
+                }}
+              />
+            </div>
+          </div>
+          <div className="roles-top-right">
+            <Button
+              className="role-add"
+              color="primary"
+              endIcon={<AddIcon />}
+              onClick={() => {
+                setShowEditRole(true);
+                setEditedRole({});
               }}
-            />
+            >
+              Add Role
+            </Button>
           </div>
-          {/* <Button color="primary" onClick={() => setShowAddRole(true)}>
-            Add New Role
-          </Button> */}
         </div>
-        <div className="roles-bot">
-          {uniqueRoles.map((role) =>
```

Comment:
`uniqueRoles` is no longer being used, so the declaration on line 95 could be removed.